FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataSource.cs

```diff
@@ -18,7 +18,10 @@ public FhirApiDataSource(IOptions<FhirServerConfiguration> config)
             EnsureArg.IsNotNull(config, nameof(config));
             EnsureArg.IsNotNullOrEmpty(config.Value.ServerUrl, nameof(config.Value.ServerUrl));
 
-            FhirServerUrl = config.Value.ServerUrl;
+            // If the baseUri has relative parts (like /api), then the relative part must be terminated with a slash (like /api/).
+            // Otherwise the relative part will be omitted when creating new search Uris. See https://docs.microsoft.com/en-us/dotnet/api/system.uri.-ctor?view=net-6.0
+            FhirServerUrl = !config.Value.ServerUrl.EndsWith("/") ? $"{config.Value.ServerUrl}/" : config.Value.ServerUrl;
```

Comment:
Is it possible that the `FhirServerUrl` contains a pound key like "https://ferris-service.fhir.azurehealthcareapis.com#"? Seems this is also a valid FHIR server URL.

Reply:
pound key will be ignored when creating `URI` correctly.

unit tests are added in `FhirApiDataClientTests`  to check it.